### ISSUE TITLE
Add CHANGELOG/updated Documentation

### ISSUE THREAD

**Issue by Radonirinaunimi** [NNPDF/eko#216]
Since EKO is now being used in multiple projects, would it be possible to have **change logs** whenever breaking changes are added in tags/releases? For the past couple of weeks, in our [nnusf](https://github.com/NNPDF/nnusf/) project, I've often found myself  finding changes that break backward compatibility (see for example https://github.com/NNPDF/nnusf/issues/85). Or at least the corresponding tutorials [alpha_s.ipynb](https://github.com/NNPDF/eko/blob/master/doc/source/overview/tutorials/alpha_s.ipynb) should be updated.

**Comment by AleCandido** [comment]
It has always been a nice-to-have, but never top-priority.

Essentially, doing it manually requires making all developers and reviewers responsible for the updates, making the process heavier and fallible.
And since there is not a single way to translate PRs in a change-log, I always postponed the problem for better investigation.

If you find a similar project with a good enough workflow (requiring little manual effort), I'd be glad to adopt the strategy!

**Comment by felixhekhorn** [comment]
the above refers to the CHANGELOG - the documentation surely has to be maintained as much as we can ...

just to mention: of course this is closely related to #145

**Comment by AleCandido** [comment]
> the above refers to the CHANGELOG - the documentation surely has to be maintained as much as we can ...

Agreed, but we are already putting a bit of effort to maintain the docs, and contributors are more and more aware of the necessity of maintaining updated docs (I hope...). And this for sure has to be done manually.

The automated CHANGELOG proposal is just to avoid extra burden. If we have too many things to maintain manually, at some point none of them will be sufficiently important any longer...

**Comment by Radonirinaunimi** [comment]
I fully understand that (unfortunately) in all cases some efforts will be needed. But as @AleCandido alluded there might be some ways to alleviate this burden. The best one I know for example is [github-changelog-generator](https://github.com/github-changelog-generator/github-changelog-generator), which also has been implemented in various github action for further ease.

**Comment by AleCandido** [comment]
> The best one I know for example is [github-changelog-generator](https://github.com/github-changelog-generator/github-changelog-generator), which also has been implemented in various github action for further ease.

Not very maintained in the last year, but sufficiently popular. It seems stable enough to fit the purpose.
Can you also point out one of these actions?

**Comment by Radonirinaunimi** [comment]
The actions that I was alluding to are even less maintained in the past two years 🙃 But the CLI itself should do the job (or perhaps at least worth trying, as you said it is sufficiently popular 😄).

**Comment by felixhekhorn** [comment]
The alpha_s tutorial was updated in https://github.com/NNPDF/eko/commit/45682d7775210f262529d0f639fd1d21a48b5457 - we can keep this issue for the changelog

**Comment by Radonirinaunimi** [comment]
> The alpha_s tutorial was updated in [45682d7](https://github.com/NNPDF/eko/commit/45682d7775210f262529d0f639fd1d21a48b5457) - we can keep this issue for the changelog

Awesome! Thanks.
